docs/setup/deploy-on-kubernetes.md

```diff
@@ -0,0 +1,316 @@
+#### Kubernetes-Based Deployment
+
+The Kubernetes builder extension offers native support for running Ballerina programs on Kubernetes with the use of annotations that you can include as part of your service code. Also, it will take care of the creation of the Docker images, so you don't need to explicitly create Docker images prior to deployment on Kubernetes.
+
+The following Kubernetes configurations are supported:
+- Kubernetes deployment support
+- Kubernetes service support
+- Kubernetes liveness probe support
+- Kubernetes ingress support
+- Kubernetes horizontal pod autoscaler support
+- Docker image generation
+- Docker push support with remote Docker registry
+- Kubernetes secret support
+- Kubernetes config map support
+- Kubernetes persistent volume claim support
+
+The following Ballerina code section explains how you can use some of these Kubernetes capabilities by using Kubernetes annotation support in Ballerina. 
+Full example can be found at [Database Interaction Guide](https://ballerina.io/learn/by-guide/data-backed-service/)
+
+```ballerina
```

Comment:
Can we include this code segment as described in here [1]
[1] https://github.com/wso2/ballerina-integrator/tree/master/www#b-include-a-code-segment